From 46e407d9654d5c8986b0f7b61de3aa223b1df399 Mon Sep 17 00:00:00 2001
From: Jorge Ramirez-Ortiz <jorge@foundries.io>
Date: Thu, 31 Oct 2024 15:12:04 +0100
Subject: [PATCH] CryptoPkg: Increase ScratchMemory buffer for openssl 3.0.15

Openssl 3.0.15 has a larger memory footprint.

Updating from EDK 2022.2 (openssl 1.1.j) to 2024.2 (openssl 3.0.15)
causes our EFI provisioning application[1] to fail due to an out of
memory condition.

On inspection, at the time of that fault, 2022.2 had an additional 900
pages. This is why this patch proposes the increase of the ScratchMemory
buffer by that same ammount.

[1] https://git.kernel.org/pub/scm/linux/kernel/git/jejb/efitools.git

Upstream-Status: Submitted [https://github.com/tianocore/edk2/pull/6394]

Signed-off-by: Jorge Ramirez-Ortiz <jorge@foundries.io>
Signed-off-by: Jose Quaresma <jose.quaresma@foundries.io>
---
 CryptoPkg/Library/BaseCryptLib/SysCall/RuntimeMemAllocation.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CryptoPkg/Library/BaseCryptLib/SysCall/RuntimeMemAllocation.c b/CryptoPkg/Library/BaseCryptLib/SysCall/RuntimeMemAllocation.c
index 0d2ca604ea..bb066f0dd0 100644
--- a/CryptoPkg/Library/BaseCryptLib/SysCall/RuntimeMemAllocation.c
+++ b/CryptoPkg/Library/BaseCryptLib/SysCall/RuntimeMemAllocation.c
@@ -33,7 +33,7 @@ SPDX-License-Identifier: BSD-2-Clause-Patent
 #define RT_PAGE_FREE  0x00000000
 #define RT_PAGE_USED  0x00000001
 
-#define MIN_REQUIRED_BLOCKS  600
+#define MIN_REQUIRED_BLOCKS  1100
 
 //
 // Memory Page Table
-- 
2.25.1

